src/Multicall3.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.12;
+
+/// @title Multicall3
+/// @notice Aggregate results from multiple read-only function calls
```

Comment:
Doesn't multicall support writing function calls?

Reply:
True, it does! I think read-only is the more common use case so how about
```suggestion
/// @notice Aggregate results from multiple read-only function calls or batch on-chain calls into a single transaction
```

Reply:
Updated in d2320fb